tgui/packages/tgui/interfaces/AccountingConsole.tsx

```diff
@@ -1,132 +1,614 @@
 import { useState } from 'react';
 import {
   Blink,
-  BlockQuote,
-  Collapsible,
+  Box,
+  Button,
+  DmIcon,
+  Flex,
   Modal,
+  NumberInput,
   Section,
   Stack,
-  Tabs,
+  Table,
 } from 'tgui-core/components';
 import type { BooleanLike } from 'tgui-core/react';
 
 import { useBackend } from '../backend';
 import { Window } from '../layouts';
 
 type Data = {
-  PlayerAccounts: PlayerAccount[];
-  AuditLog: AuditLog[];
-  Crashing: BooleanLike;
+  accounts: PlayerAccount[];
+  audit_log: AuditLog[];
+  crashing: BooleanLike;
+  pic_file_format: string;
+  max_pay_mod: number;
+  min_pay_mod: number;
+  max_advances: number;
+  station_time: string;
+  young_ian: BooleanLike;
 };
 
 type PlayerAccount = {
-  index: number;
+  id: number;
   name: string;
   balance: number;
   job: string;
   modifier: number;
+  num_advances: number;
 };
 
 type AuditLog = {
-  index: number;
   account: number;
   cost: number;
   vendor: string;
+  stationtime: string;
 };
 
 enum SCREENS {
+  none,
   users,
   audit,
+  ian,
 }
 
-export const AccountingConsole = (props) => {
-  const [screenmode, setScreenmode] = useState(SCREENS.users);
+export const AccountingConsole = () => {
+  const { data } = useBackend<Data>();
+  const {
+    station_time = '00:00',
+    pic_file_format = 'png',
+    young_ian = false,
+  } = data;
+  const [screenmode, setScreenmode] = useState(SCREENS.none);
+
+  const ianFileName = young_ian
+    ? `Ian's first birthday.${pic_file_format}`
+    : `Ian.${pic_file_format}`;
 
   return (
-    <Window width={300} height={360}>
-      <Window.Content>
-        <Stack fill vertical>
-          <MarketCrashing />
+    <Window width={600} height={440} theme="ntOS95">
+      <Window.Content fontFamily="Tahoma">
+        <Stack vertical fill>
           <Stack.Item>
-            <Tabs fluid textAlign="center">
-              <Tabs.Tab
-                selected={screenmode === SCREENS.users}
-                onClick={() => setScreenmode(SCREENS.users)}
-              >
-                Users
-              </Tabs.Tab>
-              <Tabs.Tab
-                selected={screenmode === SCREENS.audit}
-                onClick={() => setScreenmode(SCREENS.audit)}
-              >
-                Audit
-              </Tabs.Tab>
-            </Tabs>
+            <Flex height="355px">
+              <Flex.Item width="100px">
```

Comment:
Why flex? Can you use stack grow to create the ui so that it doesn't rely on pixel dimensions?

Reply:
I couldn't figure out how to make it sit correctly with grow.
It just caused the toolbar to jump around in size.
And if I gave the toolbar a set size instead, it would override the screen.